src/Feature/Editor/IndentLineRenderer.re

```diff
@@ -65,34 +67,28 @@ let paint = Skia.Paint.make();
 
 let render =
     (
-      ~canvasContext,
+      ~context: Draw.context,
       ~buffer: Buffer.t,
       ~startLine: int,
       ~endLine: int,
-      ~lineHeight: float,
-      ~fontWidth: float,
-      ~bufferPositionToPixel,
-      ~cursorLine,
+      ~cursorPosition: Location.t,
       ~theme: Theme.t,
-      ~indentationSettings: IndentationSettings.t,
       ~showActive: bool,
-      (),
+      indentationSettings: IndentationSettings.t,
     ) => {
   /* First, render *all* indent guides */
   let bufferLineCount = Buffer.getNumberOfLines(buffer);
+  let cursorLine = Index.toZeroBased(cursorPosition.line);
   let startLine = max(0, startLine);
   let endLine = min(bufferLineCount, endLine);
 
   let cursorLineIndentLevel = ref(0);
   let previousIndentLevel = ref(0);
 
   let indentationWidthInPixels =
-    float_of_int(indentationSettings.tabSize) *. fontWidth;
+    float(indentationSettings.tabSize) *. context.charWidth;
 
-  let l = ref(startLine);
-
-  while (l^ < endLine) {
-    let line = l^;
+  for (line in startLine to endLine - 1) {
```

Comment:
I've swapped out a few while loops that just looped over a counter to use for loops instead, since that's kind of their thing.

Reply:
There's still some while loops below here though, which uses a bunch of unnecessary `ref`s. The idiomatic way of doing a loop that aborts on a certain condition is to raise an exception when the condition is met. Which might seem silly, but exceptions are really cheap in OCaml.

Reply:
The `for` loops are cleaner, thanks for switching!